clang/lib/Frontend/Rewrite/FrontendActions.cpp

```diff
@@ -270,7 +270,7 @@ class RewriteIncludesAction::RewriteImportsListener : public ASTReaderListener {
 bool RewriteIncludesAction::BeginSourceFileAction(CompilerInstance &CI) {
   if (!OutputStream) {
     OutputStream =
-        CI.createDefaultOutputFile(/*Binary=*/true, getCurrentFileOrBufferName());
+        CI.createDefaultOutputFile(/*Binary=*/false, getCurrentFileOrBufferName());
```

Comment:
Same here: I don't think we should have to touch this code.

Reply:
This changes if the raw_fd_stream in OutputFile gets the Text attribute, which will change how takeOSImpl works.

Essentially, this will cause the output cannot be piped from stdout. There are tests in clang that uses this behavior so this is a regression.

Reply:
What I'm saying is that that takeOSImpl *should* be equivalent to the old logic in CompilerInstance.cpp. If there's a regression, we should fix takeOSImpl to have equivalent logic.

Reply:
That case should be hitting this logic:
```
    // Write out the content and close the stream.
    assert(BufferedContent && "Need to write content to a stream");
    OS->write(BufferedContent->begin(), BufferedContent->size());
    OS = nullptr;
```
which should be writing to the pipe.
- Is that logic not getting hit?
- Is `BufferedContent` missing the data?
- Is `OS->write()` not working correctly?

Reply:
(this logic should be equivalent to using a `buffer_unique_ostream`, which is what CompilerInstance.cpp had, it's just going through a bunch of other logic on the way there)

Reply:
Let me know if the current implementation works better. I teach OnDiskOutputFile to use `buffer_unique_ostream` instead of using `raw_svector_ostream`

Reply:
I guess `OnDiskOutputFile::closeFile` is not even hit. I am not sure what you were planned to call it?

Reply:
Looks to me like `OnDiskOutputFile::closeFile` is called by both `OnDiskOutputFile::storeContentBuffer()` and `::storeStreamedContent()`, and `OutputFile::close()` should call at least one of them. Is `OutputFile::close()` not getting called?